pytorch_lightning/utilities/parsing.py

```diff
@@ -204,24 +212,41 @@ def lightning_getattr(model, attribute):
             attr = model.hparams[attribute]
         else:
             attr = getattr(model.hparams, attribute)
+    elif hasattr(model.datamodule, attribute):
+        attr = getattr(model.datamodule, attribute)
+    elif trainer is not None and trainer.datamodule is not None and hasattr(trainer.datamodule, attribute):
```

Comment:
Why is it we both need to check that `model.datamodule` and `trainer.datamodule`, are these different?

Reply:
They should not be different. This is probably a side effect of allowing setting the `datamodule` in both the model and the trainer. Looking at the actual attach code https://github.com/PyTorchLightning/pytorch-lightning/blob/ee72271d205beb2c012e34425cb16e189cc56c7d/pytorch_lightning/trainer/data_connector.py#L96-L116
model.datamodule seems to take precedence.

Reply:
Alright I get it now, `trainer.datamodule` will exist in both the case where datamodule is defined as part of the model or if it is given to `fit` directly. Therefore, we should probably only check if `trainer.datamodule` has the attribute and alter that.

Reply:
ok, thanks for bringing this up. I also see now that in the case where module.datamodule is defined and fit receives a DIFFERENT datamodule, we should only update the one passed to fit because that's the one that take precedence. 
I'll make the changes

Reply:
@SkafteNicki actually, just found out that the scaling methods never pass the dataloaders/datamodule to fit. 
This means tuning currently only works with dataloaders and datamodules defined on the LightningModule itself. 
We should either remove the dataloader args from .tune() or add this functionality. 

To make this PR managable, I propose to go with former and properly add the functionality in a new PR. What do you think?

Reply:
IMO, we should only support passing the datamodule to `.{fit,test,tune}...` instead of also to the model. Models should be independent of the datamodules. It would also avoid the confusion of having both ways to do it.

Reply:
@carmocca not sure what you mean, without passing model to the fit method, what do we fit to the data? Anyway, I think it's a different discussion.

I solved the problem by passing all args passed to .tune() down to the batch_scaling.

Reply:
Sorry, the way I worded it makes it ambiguous. What I meant is that we support both:
```
dm = MyLightningDataModule()
model = MyLightningModule(dm)
trainer.fit(model)
```
and
```
dm = MyLightningDataModule()
model = MyLightningModule()
trainer.fit(model, datamodule=dm)
```
But should only support the second option.

It is a different discussion, just wanted to pick your brain :smile:

Reply:
ok. that is something to discuss. however, currently we need both because datamodules are new and before we only had dataloader args. for now, I think the fix here is necessary if we want to support datamodule in batch- and lrfinder